pkg/dns/aws/dns.go

```diff
@@ -101,10 +106,14 @@ type ServiceEndpoint struct {
 }
 
 func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error) {
-	sess, err := session.NewSessionWithOptions(session.Options{
+	sessionOpts := session.Options{
 		SharedConfigState: session.SharedConfigEnable,
 		SharedConfigFiles: []string{config.SharedCredentialFile},
-	})
+	}
+	if config.CustomCABundle != "" {
+		sessionOpts.CustomCABundle = strings.NewReader(config.CustomCABundle)
```

Comment:
In the event that `config.CustomCABundle` is empty when it should not be (ie on a c2s cluster), will `NewSessionWithOptions` return an error? Or where would we error out in that case?

Reply:
If the custom CA bundle is not present but required, the `NewSessionWithOptions` call will still succeed. That call does not actually make any calls to the AWS API. The errors will come in the AWS API calls (eg, the `ListHostedZonesPages` call). The errors will be x509 errors and look something like the following.
```
RequestError: send request failed
caused by: Post \"https:/route53.c2s.ic.gov/\": x509: certificate signed by unknown authority" 
```

Reply:
Ultimately, you will see this as a failure to create the DNS provider, with an error from the call to `validateServiceEndpoints`.